src/domains/Login/_components/Form.tsx

```diff
@@ -1,39 +1,65 @@
 import React from 'react';
 import { useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useDispatch } from 'react-redux';
+import { useHistory } from 'react-router';
 
 import { Grid } from '@material-ui/core';
 
 import { Container, Button, Input } from './Form.styled';
 
+import { loginUser } from '@state/_redux/user/actions';
+import paths from '@shared/paths';
+
+interface IProps {}
+
 interface ILoginData {
     email: string;
     password: string;
 }
 
-const Form = () => {
+const schema = yup.object().shape({
```

Comment:
Move it to validation file. Moreover, we can create validations in shared folder with duplicated vaildations like email, username